src/api.cc

```diff
@@ -89,6 +90,16 @@ AkliteClient::AkliteClient(const boost::program_options::variables_map& cmdline_
   Init(config);
 }
 
+AkliteClient::~AkliteClient() {
+  // Release the lock to allow reobtaining with another instance.
+  int fd = open("/var/lock/aklite.lock", O_CREAT | O_RDONLY, 0444);
+  if (fd >= 0) {
+    flock(fd, LOCK_UN);
+    close(fd);
+    unlink("/var/lock/aklite.lock");
+  }
```

Comment:
I bet this will all work by just calling `unlink("/var/lock/aklite.lock");` the active process isn't dropping the original flock, its just causing us to flock a new file/inode

Reply:
Updated; I think you're right. I'll test shortly.